File: lua/unit_test/demo/spec/calculator_spec.lua
local calculator = require("calculator")

describe("Calculator", function()
  it("adds two numbers correctly", function()
    assert.are.equal(5, calculator.add(2, 3))
  end)

  it("subtracts two numbers correctly", function()
    assert.are.equal(3, calculator.subtract(5, 2))
  end)
end)

File: _experimental/_haskell/unit_test/demo/test/calculator_spec.hs
import Test.Hspec
import Calculator (addition, subtraction)

main :: IO ()
main = hspec $ do
  describe "Calculator" $ do
    describe "add" $ do
      it "adds two positive numbers" $ do
        addition 2 3 `shouldBe` 5

    describe "subtract" $ do
      it "subtracts two positive numbers" $ do
        subtraction 5 3 `shouldBe` 2